did_peer/src/key/key.rs

```diff
@@ -0,0 +1,244 @@
+use crate::error::DidPeerError;
+
+use super::SupportedKeyType;
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Key {
+    key_type: SupportedKeyType,
+    key: Vec<u8>,
+}
+
+impl Key {
+    pub fn new(key: Vec<u8>, key_type: SupportedKeyType) -> Result<Self, DidPeerError> {
+        // If the key is a multibase key coming from a verification method, for some reason it is also
+        // multicodec encoded, so we need to strip that. But should it be?
+        let key = Self::strip_multicodec_prefix_if_present(key, &key_type);
+        Ok(Self { key_type, key })
+    }
+
+    pub fn key_type(&self) -> &SupportedKeyType {
+        &self.key_type
+    }
+
+    pub fn key(&self) -> &[u8] {
+        self.key.as_ref()
+    }
+
+    pub fn prefixed_key(&self) -> Vec<u8> {
```

Comment:
This is a [multicodec](https://github.com/multiformats/multicodec) prefix. The multicodec codes are encoded as unsinged varints (variable-width integers). The [maximum practical size of a varint is 63 bits](https://github.com/multiformats/unsigned-varint#practical-maximum-of-9-bytes-for-security), which is 9 bytes (the first bit of each byte is a continuation bit), which about fits into `u64`. I don't know why the length of the buffer 10 instead of 9, but it's a constant defined by the library.

Reply:
Thank you for clarifying, then how about calling it explicitly `multicoded_prefixed_key`? Or else leaving a comment about what sort of prefix it is

Reply:
Sounds good